app/models/mixins/authentication_mixin.rb

```diff
@@ -129,24 +129,17 @@ def update_authentication(data, options = {})
 
     data.each_pair do |type, value|
       cred = authentication_type(type)
-      current = {:new => nil, :old => nil}
-
-      unless value.key?(:userid) && value[:userid].blank?
-        current[:new] = {:user => value[:userid], :password => value[:password], :auth_key => value[:auth_key]}
-      end
-      current[:old] = {:user => cred.userid, :password => cred.password, :auth_key => cred.auth_key} if cred
 
       # Raise an error if required fields are blank
       Array(options[:required]).each { |field| raise(ArgumentError, "#{field} is required") if value[field].blank? }
 
+      attributes = [:userid, :password, :auth_key]
```

Comment:
You might not want to use the variable name `attributes` because it shadows the common method `attributes` from ActiveRecord::Base.